### ISSUE TITLE
Pagination problem and deployment 

### ISSUE THREAD

**Issue by Sarvesh001112** [argoproj/argo-cd#21023]
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```
Dear Argo CD Support Team,

 I am encountering two issues with the Argo CD API: 

1. Pagination Issue: When I make a request to:https://localhost:8083/api/v1/applications?limit=10I am receiving all responses, rather than the expected 10 results. Despite setting the limit parameter to 10, all items are being returned. 
In short - limit parameter not working

2. Deployment Issue: When I query the API using:https://localhost:8083/api/v1/applications?name={application name}the deployments from two days ago are not appearing, while recent ones from 2023 are. 

Could you please assist me in resolving these issues or provide guidance on proper pagination and deployment retrieval handling for these endpoints? 
Thank you for your support. 
Best regards, 
Sarvesh

**Comment by nitishfy** [comment]
> Pagination Issue: When I make a request to:https://localhost:8083/api/v1/applications?limit=10I am receiving all responses, rather than the expected 10 results. Despite setting the limit parameter to 10, all items are being returned.
In short - limit parameter not working

pagination 'may' require both `limit` and `continue` parameters. If `continue` is missing and the API call includes more items than the limit, it might return all results, bypassing pagination. You should check the response headers for the continue token if the results are paginated.

**Comment by Sarvesh001112** [comment]
But continue will come in response at end of json . If all response is coming how can we get continue. Continue will used in 2 time

**Comment by Sarvesh001112** [comment]
I want to run api and want only 10 response at 1st time . Also what is the default max value of limit.

**Comment by Sarvesh001112** [comment]
Hi support team,
 
Request you to please share update?

**Comment by Sarvesh001112** [comment]
I am only seeing 10 deployment in 1 application not more than that why?


If we deploy more than 10 - let say for 1st - I'd =1 and so on for 10 I'd = 10 .

If I deploy 11 application than 1st is gone and in response I'd =2 to I'd =11 is coming . 


Is there any limit that only 10 deployment is showing.
